src/Quartz.Extensions.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -113,23 +113,37 @@ public static class ServiceCollectionExtensions
             JobKey? jobKey = null,
             Action<IJobConfigurator>? configure = null) where T : IJob
         {
-            var c = new JobConfigurator();
-            if (jobKey != null)
+            return AddJob(options,typeof(T), jobKey, configure);
+        }
+        /// <summary>
+        /// Add job to underlying service collection.jobType shoud be implement `IJob`
+        /// </summary>
+        public static IServiceCollectionQuartzConfigurator AddJob(
+           this IServiceCollectionQuartzConfigurator options,
+           Type jobType,
+           JobKey? jobKey = null,
+           Action<IJobConfigurator>? configure = null)
+        {
+            if (typeof(IJob).IsAssignableFrom(jobType))
```

Comment:
I think there it's an error condition if the job doesn't implement `IJob`, should throw `ArgumentException` as pre-check, will make the diff smaller also (less nested).